.github/actions/action.yml

```diff
@@ -0,0 +1,38 @@
+name: 'Run Tests'
+description: 'Run Tests'
+inputs:
+  repo-name:
+    description: 'Repository to be tested'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Clone mtreinish/qiskit-neko PUBLIC repository
+      uses: mtreinish/clone-github-repo-action@v2
+      with:
+        owner: 'mtreinish'
+        repository: 'qiskit-neko'
+    - name: Access cloned repository content
+      run: |
+        cd qiskit-neko
+    - name: Download Released Dependencies
+      run: |
+        pip install .
+        cd ..
+    - name: Clone qiskit/"${{ inputs.repo-name }}" PUBLIC repository
+      uses: Qiskit/clone-github-repo-action@v2
+      with:
+        owner: 'Qiskit'
+        repository: "${{ inputs.repo-name }}"    
```

Comment:
We probably should use actions/checkout https://github.com/actions/checkout here for this step. I think the expected workflow here we would want is to have qiskit-terra define a job that's like:

```yaml
---
steps:
  - name: Run Neko
    uses: mtreinish/qiskit-neko
    with:
      tests: terra
```
that will run on proposed changes in pull requests. Using the checkout action will mean the job will have the proposed changes from a PR when we install it.